src/parquet/encodings/dictionary-encoding.h

```diff
@@ -360,7 +367,12 @@ inline void DictEncoder<T>::DoubleTableSize() {
 
   hash_table_size_ = new_size;
   mod_bitmask_ = new_size - 1;
-  new_hash_slots.swap(hash_slots_);
+
+  hash_slots_.Clear();
+  hash_slots_.Reserve(hash_table_size_);
+  for (int i = 0; i < hash_table_size_; i++) {
+    hash_slots_[i] = new_hash_slots[i];
+  }
```

Comment:
This has a distinctly different performance profile than using `std::vector::swap`. Perhaps we should implement `swap` for our `Vector`?